Error when putting <auth> in pom.xml

From https://github.com/GoogleContainerTools/jib/issues/808#issuecomment-411225404

This worked in 0.9.8, so this is a regression.
```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.9-SNAPSHOT:build (default-cli) on project helloworld: Unable to parse configuration of mojo com.google.cloud.tools:jib-maven-plugin:0.9.9-SNAPSHOT:build for parameter auth: Cannot create instance of class com.google.cloud.tools.jib.maven.JibPluginConfiguration$AuthConfiguration: com.google.cloud.tools.jib.maven.JibPluginConfiguration$AuthConfiguration.<init>() -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginConfigurationException
```
Gradle works OK though.

--- a/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java
+++ b/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java
@@ -68,11 +68,6 @@ abstract class JibPluginConfiguration extends AbstractMojo {
       return password;
     }
 
-    void setPropertyDescriptors(String usernameDescriptor, String passwordDescriptor) {
-      this.usernameDescriptor = usernameDescriptor;
-      this.passwordDescriptor = passwordDescriptor;
-    }
-
     @VisibleForTesting
     void setUsername(String username) {
       this.username = username;
@@ -81,6 +76,11 @@ abstract class JibPluginConfiguration extends AbstractMojo {
     @VisibleForTesting
     void setPassword(String password) {
       this.password = password;
+    }
+
+    private void setPropertyDescriptors(String usernameDescriptor, String passwordDescriptor) {
+      this.usernameDescriptor = usernameDescriptor;
+      this.passwordDescriptor = passwordDescriptor;
     }
   }
 

--- a/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java
+++ b/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java
@@ -43,32 +43,17 @@ abstract class JibPluginConfiguration extends AbstractMojo {
 
     @Nullable @Parameter private String username;
     @Nullable @Parameter private String password;
-    private String usernameDescriptor;
-    private String passwordDescriptor;
-
-    private AuthConfiguration(String descriptor) {
-      this.usernameDescriptor = descriptor + "<username>";
-      this.passwordDescriptor = descriptor + "<password>";
-    }
-
-    @VisibleForTesting
-    void setUsername(String username) {
-      this.username = username;
-    }
-
-    @VisibleForTesting
-    void setPassword(String password) {
-      this.password = password;
-    }
+    @Nullable private String usernameDescriptor;
+    @Nullable private String passwordDescriptor;
 
     @Override
     public String getUsernamePropertyDescriptor() {
-      return usernameDescriptor;
+      return Preconditions.checkNotNull(usernameDescriptor);
     }
 
     @Override
     public String getPasswordPropertyDescriptor() {
-      return passwordDescriptor;
+      return Preconditions.checkNotNull(passwordDescriptor);
     }
 
     @Override
@@ -81,6 +66,21 @@ abstract class JibPluginConfiguration extends AbstractMojo {
     @Nullable
     public String getPassword() {
       return password;
+    }
+
+    void setPropertyDescriptors(String usernameDescriptor, String passwordDescriptor) {
+      this.usernameDescriptor = usernameDescriptor;
+      this.passwordDescriptor = passwordDescriptor;
+    }
+
+    @VisibleForTesting
+    void setUsername(String username) {
+      this.username = username;
+    }
+
+    @VisibleForTesting
+    void setPassword(String password) {
+      this.password = password;
     }
   }
 
@@ -96,7 +96,7 @@ abstract class JibPluginConfiguration extends AbstractMojo {
 
     @Nullable @Parameter private String credHelper;
 
-    @Parameter private AuthConfiguration auth = new AuthConfiguration("<from><auth>");
+    @Parameter private AuthConfiguration auth = new AuthConfiguration();
   }
 
   /** Configuration for {@code to} parameter, where image is required. */
@@ -106,7 +106,7 @@ abstract class JibPluginConfiguration extends AbstractMojo {
 
     @Nullable @Parameter private String credHelper;
 
-    @Parameter private AuthConfiguration auth = new AuthConfiguration("<to><auth>");
+    @Parameter private AuthConfiguration auth = new AuthConfiguration();
 
     public void set(String image) {
       this.image = image;
@@ -167,6 +167,12 @@ abstract class JibPluginConfiguration extends AbstractMojo {
   private String extraDirectory;
 
   @Nullable @Component protected SettingsDecrypter settingsDecrypter;
+
+  /** Default constructor handles setting up auth property descriptors. */
+  JibPluginConfiguration() {
+    to.auth.setPropertyDescriptors("<to><auth><username>", "<to><auth><password>");
+    from.auth.setPropertyDescriptors("<from><auth><username>", "<from><auth><password>");
+  }
 
   /**
    * Warns about deprecated parameters in use.

--- a/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java
+++ b/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java
@@ -78,9 +78,9 @@ abstract class JibPluginConfiguration extends AbstractMojo {
       this.password = password;
     }
 
-    private void setPropertyDescriptors(String usernameDescriptor, String passwordDescriptor) {
-      this.usernameDescriptor = usernameDescriptor;
-      this.passwordDescriptor = passwordDescriptor;
+    private void setPropertyDescriptors(String descriptorPrefix) {
+      this.usernameDescriptor = descriptorPrefix + "<username>";
+      this.passwordDescriptor = descriptorPrefix + "<password>";
     }
   }
 
@@ -170,8 +170,8 @@ abstract class JibPluginConfiguration extends AbstractMojo {
 
   /** Default constructor handles setting up auth property descriptors. */
   JibPluginConfiguration() {
-    to.auth.setPropertyDescriptors("<to><auth><username>", "<to><auth><password>");
-    from.auth.setPropertyDescriptors("<from><auth><username>", "<from><auth><password>");
+    to.auth.setPropertyDescriptors("<to><auth>");
+    from.auth.setPropertyDescriptors("<from><auth>");
   }
 
   /**

--- a/jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/JibPluginConfigurationTest.java
+++ b/jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/JibPluginConfigurationTest.java
@@ -20,6 +20,7 @@ import com.google.cloud.tools.jib.JibLogger;
 import java.nio.file.Paths;
 import java.util.Arrays;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
@@ -32,14 +33,35 @@ public class JibPluginConfigurationTest {
 
   @Mock private JibLogger mockLogger;
 
-  @Test
-  public void testHandleDeprecatedParameters() {
-    JibPluginConfiguration testPluginConfiguration =
+  private JibPluginConfiguration testPluginConfiguration;
+
+  @Before
+  public void setup() {
+    testPluginConfiguration =
         new JibPluginConfiguration() {
           @Override
           public void execute() {}
         };
+  }
 
+  @Test
+  public void testAuthDefaults() {
+    Assert.assertEquals(
+        "<from><auth><username>",
+        testPluginConfiguration.getBaseImageAuth().getUsernamePropertyDescriptor());
+    Assert.assertEquals(
+        "<from><auth><password>",
+        testPluginConfiguration.getBaseImageAuth().getPasswordPropertyDescriptor());
+    Assert.assertEquals(
+        "<to><auth><username>",
+        testPluginConfiguration.getTargetImageAuth().getUsernamePropertyDescriptor());
+    Assert.assertEquals(
+        "<to><auth><password>",
+        testPluginConfiguration.getTargetImageAuth().getPasswordPropertyDescriptor());
+  }
+
+  @Test
+  public void testHandleDeprecatedParameters() {
     testPluginConfiguration.handleDeprecatedParameters(mockLogger);
     Mockito.verify(mockLogger, Mockito.never()).warn(Mockito.any());
 

